### PR TITLE
[POC] Change Baseimage to Distroless Node Image

### PR DESCRIPTION
**Pull request by kaindlnetwork**
Hello everybody,

We have again some change ideas to make this project a bit more awesome. The current goal needs some testing, but we want to get away from alpine with its own package manager. The reason is if somebody gains access inside the container it is very likely to use the package manager and packages like wget and curl to attack your infrastructure from the inside. We want to prevent this directly by removing the possibility to add new packages.

tldr:
[POC] Reduce attack surface as there is no package manager and reduce the image size further

Further Sources:
https://snyk.io/blog/choosing-the-best-node-js-docker-image/

### CODE REVIEW

**Comment by schlagmichdoch** [comment]
This sounds great but honestly, I do not know enough to have an informed opinion about such changes.

Also, it would be good to have the same base image for all provided docker images including the one hosted by linuxserver: https://github.com/linuxserver/docker-pairdrop

@thespad As you have created the linuxserver docker image and you know a lot about docker images apparently, I would appreciate your view on this! Do you think this change would be sensible? Would you be open to implementing this change into the linuxserver image as well?

**Comment by thespad** [comment]
So, the short answer is no, we can't do it with the Linuxserver images as it stands, for a number of reasons including breaking functionality that we specifically offer people (mods and custom scripts).

That said, all the application processes in our image run as a non-root user (unless you specifically force it otherwise, which we don't support) so someone compromising the application couldn't install any additional packages using the package manager. They could make use of installed "core" packages like curl, but would still be limited to running anything as an unprivileged user.

Distroless images are an interesting concept (although notably are *not* the same as building from `scratch` and do include a bunch of other packages to provide a working environment for whatever type of application you're running), but not something that really works for our use case.

If you think a distroless image is of value to your users, or to you, we won't be offended if you want to offer one, or get a 3rd party to create one for you. Giving people more options is rarely a bad thing, as long as they're all properly maintained and supported.
